jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/rtree/LeafNode.java

```diff
@@ -296,8 +295,7 @@ public T getPickedObject(Point2D p) {
    * @return a subset of elements whose bounds intersect with the passed shape
    */
   @Override
-  public Collection<T> getVisibleElements(Shape shape) {
-    Collection<T> visibleElements = Sets.newHashSet();
+  public Collection<T> getVisibleElements(Collection<T> visibleElements, Shape shape) {
```

Comment:
why pass in visibleElements rather than just constructing and returning a `Set`?